istio-control/istio-discovery/templates/deployment.yaml

```diff
@@ -101,30 +114,33 @@ spec:
           volumeMounts:
           - name: config-volume
             mountPath: /etc/istio/config
-
-
-       ### Sidercar - should only be used if MTLS is enabled.
-{{- if .Values.global.controlPlaneSecurityEnabled }}
-
         - name: istio-proxy
 {{- if contains "/" .Values.global.proxy.image }}
           image: "{{ .Values.global.proxy.image }}"
 {{- else }}
-          image: "{{ .Values.global.hub | default "gcr.io/istio-release" }}/{{ .Values.global.proxy.image | default "proxyv2" }}:{{ .Values.global.tag | default "release-1.1-latest-daily" }}"
+          image: "{{ .Values.global.hub }}/{{ .Values.global.proxy.image }}:{{ .Values.global.tag }}"
 {{- end }}
           imagePullPolicy: {{ .Values.global.imagePullPolicy }}
           ports:
           - containerPort: 15011
           args:
           - proxy
+          - --domain
+          - $(POD_NAMESPACE).svc.{{ .Values.global.proxy.clusterDomain }}
           - --serviceCluster
           - istio-pilot
           - --templateFile
           - /var/lib/envoy/envoy.yaml.tmpl
+        {{- if .Values.global.controlPlaneSecurityEnabled}}
```

Comment:
This was also removed intentionally - has no effect ( we can remove the simple flag as well ).